src/ConnectedMode/Migration/IConnectedModeMigration.cs

```diff
@@ -71,4 +62,49 @@ public MigrationProgress(int currentProject, int totalProjects, string message,
         /// </summary>
         public bool IsWarning { get; }
     }
+
+    /// <summary>
+    /// Finds the set of MSBuild files that need to be cleaned as part of the migration
+    /// </summary>
+    internal interface IFileProvider
+    {
+        /// <summary>
+        /// Returns a list of full paths to the set of MSBuild files to be cleaned
+        /// </summary>
+        /// <remarks>The files can be project files or imported props/targets etc</remarks>
+        Task<IEnumerable<string>> GetFilesAsync(CancellationToken token);
+    }
+
+    /// <summary>
+    /// Contract for a component that will clean ruleset and additional file references for a single filee
+    /// </summary>
+    /// <remarks>The file will be an MSBuild file i.e. XML. It could be project file/imported prop or targets/
+    /// Directory.Build.props/targets</remarks>
+    internal interface IFileCleaner
+    {
+        Task CleanAsync(string filePath, LegacySettings legacySettings, CancellationToken token);
+    }
+
+    /// <summary>
+    /// Data class containing information about the paths to the legacy generated files
+    /// </summary>
+    /// <remarks>Required by <see cref="IFileCleaner"/> to identify the references to remove</remarks>
+    internal class LegacySettings
```

Comment:
I added this data class - the cleaner is going to need to know what strings to look for.